app-modules/segment/src/Actions/BulkSegmentAction.php

```diff
@@ -0,0 +1,101 @@
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Segment\Actions;
+
+use Exception;
+use Filament\Forms\Form;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use AdvisingApp\Segment\Models\Segment;
+use Filament\Forms\Components\Textarea;
+use Filament\Tables\Actions\BulkAction;
+use Filament\Forms\Components\TextInput;
+use Filament\Notifications\Notification;
+use AdvisingApp\Segment\Enums\SegmentType;
+use AdvisingApp\Segment\Enums\SegmentModel;
+
+class BulkSegmentAction
+{
+    public static function make(SegmentModel $segmentModel)
+    {
+        return BulkAction::make('segment')
+            ->icon('heroicon-o-rectangle-group')
+            ->label('Create Segment')
+            ->form(fn (Form $form) => $form->schema([
+                TextInput::make('name')
+                    ->label('Name')
+                    ->required()
+                    ->maxLength(255),
+
+                Textarea::make('description')
+                    ->label('Description')
+                    ->maxLength(65535),
+            ]))
+            ->action(function (Collection $records, array $data) use ($segmentModel) {
+                try {
+                    DB::beginTransaction();
+                    $data['type'] = SegmentType::Static;
+                    $data['filters'] = [];
+                    $data['model'] = $segmentModel;
+                    $segment = Segment::create($data);
+                    $id = $segmentModel->getLabel() == 'Student' ? 'sisid' : 'id';
+                    $records->pluck($id)->chunk(100)->each(function ($idChunk) use ($segment, $segmentModel) {
+                        $subjectData = $idChunk->map(fn ($id) => [
+                            'subject_id' => $id,
+                            'subject_type' => $segmentModel,
+                        ])->toArray();
+                        $segment->subjects()->createMany($subjectData);
+                    });
```

Comment:
I think you can hold off on doing the pluck here so that you can make use of the full Model to properly setup the subject details.
I would just chunk these by 100, and then do the mapping like you are doing starting on like `77` but `subject_id` should be set with `$record->getKey()` and `subject_type` set with `$record->getMorphClass()`.

That would be better programmatically and easer to follow. Also in this current state, type is not exactly right as it needs to be the morph map name.

Reply:
Done, please check.